lib/maze/aws_public_ip.rb

```diff
@@ -0,0 +1,51 @@
+module Maze
+  # Determines the public IP address and port when running on Buildkite with the Elastic CI Stack for AWS
+  class AwsPublicIp
+    attr_reader :address
+
+    def initialize
+      # This class is only relevant on Buildkite
+      return unless ENV['BUILDKITE']
+
+      ip = determine_public_ip
+      port = determine_public_port
+
+      @address = "#{ip}:#{port}"
+    end
+
+    # Determines the public IP address of the running AWS instance
+    def determine_public_ip
+      `curl --silent -XGET http://169.254.169.254/latest/meta-data/public-ipv4`
```

Comment:
Is this a fixed address, and should it be parameterised and made configurable if not?

Reply:
It is fixed, but it's a good point - I should have included a comment as to what it is.  I
'll do that (see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html in the meantime).